rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCostAnnotator.scala

```diff
@@ -0,0 +1,221 @@
+    settings: KubernetesSettings,
+    apiToken: String,
+    podNamespace: String,
+    costSettings: PodDeletionCostSettings)
+    extends Actor
+    with ActorLogging
+    with Timers {
+  private val cluster = Cluster(context.system)
+  private val http = Http()(context.system)
+
+  Cluster(context.system).subscribe(context.self, classOf[ClusterEvent.MemberUp], classOf[ClusterEvent.MemberRemoved])
+
+  private lazy val sslContext = {
+    val certificates = PemManagersProvider.loadCertificates(settings.apiCaPath)
+    val factory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
+    val keyStore = KeyStore.getInstance("PKCS12")
+    keyStore.load(null)
+    factory.init(keyStore, Array.empty)
+    val km: Array[KeyManager] = factory.getKeyManagers
+    val tm: Array[TrustManager] =
+      PemManagersProvider.buildTrustManagers(certificates)
+    val random: SecureRandom = new SecureRandom
+    val sslContext = SSLContext.getInstance("TLSv1.2")
+    sslContext.init(km, tm, random)
+    sslContext
+  }
+  private val clientSslContext: Option[HttpsConnectionContext] = if (settings.secure) Some(ConnectionContext.httpsClient(sslContext)) else None
+
+  implicit val dispatcher: ExecutionContextExecutor = context.system.dispatcher
+  def receive = onMessage(0, SortedSet.empty(Member.ageOrdering))
+
+  private def onMessage(deletionCost: Int, membersByAgeDesc: SortedSet[Member]): Receive = {
+    case cs @ ClusterEvent.CurrentClusterState(members, _, _, _, _) =>
+      log.debug("Received CurrentClusterState {}", cs)
+      updateIfNewCost(deletionCost, membersByAgeDesc ++ members) // ordering used is from the first operand (so, by age)
+
+    case ClusterEvent.MemberUp(m) =>
+      log.debug("Received MemberUp {}", m)
+      updateIfNewCost(deletionCost, membersByAgeDesc + m)
+
+    case ClusterEvent.MemberRemoved(m, _) =>
+      log.debug("Received MemberRemoved {}", m)
+      updateIfNewCost(deletionCost, membersByAgeDesc - m)
+
+    case PodAnnotated =>
+      log.debug("Annotation updated successfully to {}", deletionCost)
+      // cancelling an eventual retry in case the annotation succeeded in the meantime
+      timers.cancel(RetryTimerId)
+
+    case ScheduleRetry(ex) =>
+      log.error(s"Failed to update annotation: [$ex]. Retrying with fixed delay of {}.", costSettings.retryDelay)
+      timers.startSingleTimer(RetryTimerId, RetryAnnotate, costSettings.retryDelay)
+      context.become(underRetryBackoff(deletionCost, membersByAgeDesc))
+
+    case GiveUp(er: String) =>
+      log.error(
+        "There was a client error when trying to set pod-deletion-cost annotation. " +
+        "Not retrying, check configuration. Error: {}",
+        er)
+
+    case es => log.debug("Ignoring message {}", es)
+  }
+
+  private def underRetryBackoff(deletionCost: Int, membersByAgeDesc: SortedSet[Member]): Receive = {
+    case cs@ClusterEvent.CurrentClusterState(members, _, _, _, _) =>
```

Comment:
I don't think CurrentClusterState can happen here. It will be the first message, before any MemberUp or MemberRemoved. It's only sent once.